client/editorQueryEvaluator.ts

```diff
@@ -0,0 +1,221 @@
+import { ClientContext, sendJSQuery, sendSparql, sendXQuery, sendRows } from './marklogicClient';
+import { ClientResponseProvider } from './clientResponseProvider';
+import { contentType, RowsResponse } from 'marklogic';
+import * as ml from 'marklogic';
+
+export enum EditorQueryType {
+    JS,
+    XQY,
+    SPARQL,
+    SQL,
+    ROWS
+}
+
+export class EditorQueryEvaluator {
+    private static FOPTIONS = { tabSize: 2, insertSpaces: true };
+    private static FCOMMAND = 'vscode.executeFormatDocumentProvider';
+    private static SJS = 'sjs';
+    private static XQY = 'xqy';
+
+    private context: ExtensionContext;
+    private provider: ClientResponseProvider;
+
+    public constructor(context: ExtensionContext, provider: ClientResponseProvider) {
+        this.context = context;
+        this.provider = provider;
+    }
+
+    // Exported only for testing purposes
+    public static getQueryTextFromEditor(editor: TextEditor): string {
+        const selection = editor.selection;
+        const selectionText = editor.document.getText(selection);
+        let actualQuery: string = null;
+        if (selectionText) {
+            actualQuery = selectionText;
+        } else {
+            actualQuery = editor.document.getText();
+        }
+        return actualQuery;
+    }
+
+    public editorQuery(
+        editorQueryType: EditorQueryType, editor: TextEditor, rowsResponseFormat?: ml.RowsResponseFormat
+    ): void {
+        const actualQuery = EditorQueryEvaluator.getQueryTextFromEditor(editor);
+        const cfg: WorkspaceConfiguration = workspace.getConfiguration();
+        let language = EditorQueryEvaluator.SJS;
+        if (editorQueryType === EditorQueryType.XQY) {
+            language = EditorQueryEvaluator.XQY;
+        }
+        const client: ClientContext = cascadeOverrideClient(actualQuery, language, cfg, this.context.globalState);
+        const host = client.params.host;
+        const port = client.params.port;
+        const resultsEditorTabIdentifier = ClientResponseProvider.encodeLocation(editor.document.uri, host, port);
+
+        switch (editorQueryType) {
+        case EditorQueryType.XQY:
+            this.editorXQuery(client, actualQuery, resultsEditorTabIdentifier, editor);
+            break;
+        case EditorQueryType.JS:
+            this.editorJSQuery(client, actualQuery, resultsEditorTabIdentifier, editor);
+            break;
+        case EditorQueryType.SPARQL:
+            this.editorSparqlQuery(client, actualQuery, resultsEditorTabIdentifier, editor);
+            break;
+        case EditorQueryType.SQL:
+            this.editorSqlQuery(client, actualQuery, resultsEditorTabIdentifier, editor, cfg);
+            break;
+        case EditorQueryType.ROWS:
+            this.editorRowsQuery(client, actualQuery, resultsEditorTabIdentifier, editor, rowsResponseFormat);
+            break;
+        }
+    }
+
+    private editorJSQuery(
+        db: ClientContext,
```

Comment:
For new code, let's so `clientContext`, particularly as `db` is often used in the Node Client project to refer to an instance of the Node Client (which I don't like either; that happens in the Java Client project sometimes too, but I think "client" is a good name for a client object; it's of course not a database, it's a client to the database's REST API).